src/serializers/type_serializers/bytes.rs

```diff
@@ -38,8 +42,7 @@ impl TypeSerializer for BytesSerializer {
         let py = value.py();
         match value.downcast::<PyBytes>() {
             Ok(py_bytes) => match extra.mode {
-                SerMode::Json => extra
-                    .config
+                SerMode::Json => self
                     .bytes_mode
```

Comment:
Hypothesis: not looking in `extra.config` will break the arguments to `to_json`, I think we need to do `self.config.unwrap_or(extra.config.bytes_mode)`

Reply:
Figured this out - it doesn't break bc of the logic in `infer.rs` :)